File: leetcode/python/merge-strings-alternately.py
from typing import List
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        """
        i = word1
        j = word2
        """
        temp  = ""
        max_length = max(len(word1),len(word2))
        for i in range(max_length):
            if(i < len(word1)):
                temp += word1[i]
            if(i < len(word2)):
                temp += word2[i]
            i+=1
        return temp

File: leetcode/java/merge-strings-alternately.java
class Solution{
    public String mergeAlternately(String word1, String word2){
        var result = new StringBuilder();
        Integer i = 0;
        while (i<word1.length() && i < word2.length()) {
            result.append(word1.charAt(i));
            result.append(word2.charAt(i));
            i+=1;
        }
        if(i!=word1.length()){
            result.append(word1.substring(i));
        }
        if(i!=word2.length()){
            result.append(word2.substring(i));
        }
        return result.toString();
    }
}